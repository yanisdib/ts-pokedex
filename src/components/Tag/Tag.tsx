import styled, { css } from 'styled-components';


interface Props {
    children?: React.ReactNode;
    backgroundColor?: string;
}


export default styled.div<Props>`
    display: flex;
    justify-content: center;
    align-items: center;
    width: fit-content;
    min-width: 50px;
    height: fit-content;
    padding: 3px 10px;
    ${({backgroundColor}: Props) => css`
        background-image: radial-gradient(#ffffff2d 10%, transparent 10%);
        background-color: ${ backgroundColor ?? '#a8a8a8' };
    `}
    background-position: 0 0, 50px 50px;
    background-size: 5px 5px;
    font-family: var(--info-font);
    font-size: 0.8em;
    font-weight: 400;
    text-transform: uppercase;
    text-shadow: 1px 0px 1px #464444;
    color: #FFF;
    border-radius: 10px;
`;