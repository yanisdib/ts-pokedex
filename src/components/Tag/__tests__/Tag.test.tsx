import { render } from '@testing-library/react';

import Tag from '../Tag';


describe('Tag component', () => {
    test('should render without crashing', () => {
        const { container } = render(<Tag />);

        expect(container).toMatchSnapshot();
    });

    test('should render with default background color', () => {
        const {getByTestId} = render(<Tag data-testid="tag-test" />);

        expect(getByTestId('tag-test')).toHaveStyle('background-color: #A8A8A8')
    });

    test('should render with background color passed as props', () => {
        const { getByTestId } = render(
            <Tag
                data-testid="tag-test"
                backgroundColor="#CE0101"
            />
        );

        expect(getByTestId('tag-test')).toHaveStyle('background-color: #CE0101');
    });
});