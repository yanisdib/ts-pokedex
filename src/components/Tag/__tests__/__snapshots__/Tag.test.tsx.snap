// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component should render without crashing 1`] = `
<div>
  <div
    class="sc-bczRLJ gSigVM"
  />
</div>
`;
